backend/api/Services/AreaService.cs

```diff
@@ -0,0 +1,247 @@
+                return null;
+            return await _context.Decks.Where(a =>
+                a.Installation.Equals(installation) &&
+                a.Asset.Equals(asset) &&
+                a.Name.ToLower().Equals(name.ToLower())
+            ).Include(d => d.Installation).Include(i => i.Asset).FirstOrDefaultAsync();
+        }
+
+        public async Task<Installation?> ReadInstallationByAssetAndName(Asset? asset, string name)
+        {
+            if (asset == null)
+                return null;
+            return await _context.Installations.Where(a =>
+                a.Asset.Equals(asset) &&
+                a.Name.ToLower().Equals(name.ToLower())
+            ).Include(i => i.Asset).FirstOrDefaultAsync();
+        }
+
+        public async Task<Asset?> ReadAssetByName(string asset)
+        {
+            return await _context.Assets.Where(a =>
+                a.ShortName.ToLower().Equals(asset.ToLower())
+            ).FirstOrDefaultAsync();
+        }
+
+        public async Task<Area> Create(CreateAreaQuery newAreaQuery, List<Pose> safePositions)
+        {
+            var sp = new List<SafePosition>();
+            foreach (var p in safePositions)
+            {
+                sp.Add(new SafePosition(p));
+            }
+
+            var asset = await ReadAssetByName(newAreaQuery.AssetCode);
+            if (asset == null)
+            {
+                asset = new Asset
+                {
+                    Name = "", // TODO:
+                    ShortName = newAreaQuery.AssetCode
+                };
+                await _context.Assets.AddAsync(asset);
+                await _context.SaveChangesAsync();
+            }
+
+            var installation = await ReadInstallationByAssetAndName(asset, newAreaQuery.InstallationName);
+            if (installation == null)
+            {
+                installation = new Installation
+                {
+                    Asset = asset,
+                    Name = "", // TODO:
+                    ShortName = newAreaQuery.InstallationName
+                };
+                await _context.Installations.AddAsync(installation);
+                await _context.SaveChangesAsync();
+            }
+
+            var deck = await ReadDeckByAssetAndInstallationAndName(asset, installation, newAreaQuery.DeckName);
+            if (deck == null)
+            {
+                deck = new Deck
+                {
+                    Installation = installation,
+                    Asset = asset,
+                    Name = newAreaQuery.DeckName
+                };
+                await _context.Decks.AddAsync(deck);
+                await _context.SaveChangesAsync();
+            }
+
+            var existingArea = await ReadAreaByAssetAndInstallationAndDeckAndName(
+                asset, installation, deck, newAreaQuery.AreaName);
+            if (existingArea != null)
+            {
+                // TODO: maybe just append safe positions, or return an error
+            }
```

Comment:
If the area already exists do not change it at all, merely return an error. 

Updating safe positions should be a seperate API endpoint.